Source/WTF/wtf/posix/FileSystemPOSIX.cpp

```diff
@@ -270,15 +270,17 @@ std::pair<String, PlatformFileHandle> openTemporaryFile(StringView prefix, Strin
     // This is OK for now since the code using it is built on macOS only.
     ASSERT_UNUSED(suffix, suffix.isEmpty());
 
-    char buffer[PATH_MAX];
-    if (snprintf(buffer, PATH_MAX, "%s/%sXXXXXX", temporaryFileDirectory(), prefix.utf8().data()) >= PATH_MAX)
-        goto end;
+    const char* tmp = temporaryFileDirectory();
+    const char* pre = prefix.utf8().data();
```

Comment:
This is wrong. It returns a pointer to a temporary `CString`'s storage but then that temporary is deallocated, leaving the pointer deallocated. The result of `utf8()` needs to be stored in a local variable.